tests/test_integration_workflows.py

```diff
@@ -288,6 +288,20 @@ def test_training(self):
                 self.assertTrue(test_integration_value(TASK, key="output_sums", data=repeated[i][2:], rtol=1e-2))
         np.testing.assert_allclose(repeated[0], repeated[1])
 
+    @TimedCall(seconds=200, skip_timing=not torch.cuda.is_available(), daemon=False)
+    def test_timing(self):
```

Comment:
I think most of the logic in this function is duplicated to test_training, could you please combine them or extract a common internal function?